Dockerfile

```diff
@@ -36,7 +36,12 @@ ENV JRE_HOME /usr
 ENV CLASSPATH /usr/local/tomcat/bin/bootstrap.jar:/usr/local/tomcat/bin/tomcat-juli.jar
 
 WORKDIR $CATALINA_HOME
-RUN /opengrok/bin/OpenGrok deploy
+
+# custom deployment to / with redirect from /source
+RUN rm -rf /usr/local/tomcat/webapps/* && \
+    cp "/opengrok/lib/source.war" "/usr/local/tomcat/webapps/ROOT.war" && \
+    mkdir "/usr/local/tomcat/webapps/source" && \
+    echo '<% response.sendRedirect("/"); %>' > "/usr/local/tomcat/webapps/source/index.jsp"
```

Comment:
`OPENGROK_WEBAPP_CONTEXT` was what I tried first. It made absolutely no difference. The webapp is still deployed to `/source`, maybe that's a bug in OpenGrok. 

I could not see any problem with the URLs generated by the OpenGrok web interface (they all seem to be relative to where the Webapp is deployed to). Everything seems to work just fine.

If you think the OpenGrok deploy command is important, I can certainly use that and move the deployed war file to the right place.

Reply:
that is definitely a bug, I think it works for anything else than "/" ... so I guess it would be cool to fix it in OpenGrok script ... 
and yes, you're correct, there were fixes to fix the generated xrefs, so if they work, then it's even better, great! :)
based on your tests it seems OPENGROK_WEBAPP_CONTEXT  might need to be obsoleted and deleted then ...
in such case, then let's go and merge your original changes and we definitely need a bug in upstream to clarify the current situation (which seems to be even better than what I remember last time I tested this change of webapp name :)  (which is cool :-D )  )